### PR TITLE
fix: fixed hard coded numOfHeaders int, fixed letter t, fixed borderpictures & make the language switch in Settings use a dropDownMenu

### PR DESCRIPTION
**Pull request by SaFouNaldo**
## Fixes:
This PR addresses the following issues:

- Header Count Calculation: Replaced the manual incrementation of numOfHeaders with a dynamic function that calculates the total number of headers based on the current UI structure.
- Image Border Color: Updated the border color of images from blue to white to match the design specifications.
- Challenge Word List: Removed French words with special characters from the challenge list to comply with the American English alphabet and the app’s focus on ASL FingerSpelling.

## Changes Implemented:
1. Dynamic Header Calculation

- Replaced the manual increment of `numOfHeaders `with a function that dynamically computes the total size of the headers based on the defined topBar buttons and additional headers in the layout.
2. Fix Image Borders

- Updated the borders of images in the UI. Previously, the borders were displayed in blue. Now, they have been corrected to white for consistency with the intended design.
3. Language Switch Enhancement

- Modified the LanguageSwitch in the settings screen:
Clicking on the dropdown menu now displays items dynamically sourced from res/strings.
- Added explicit support for "French" and "English" while other languages are marked as "Not Supported."
No hardcoded text is used; all values are localized through resources.
4. Challenge Word List Update

- Removed French words containing special characters from the challenge word list.
Ensured that the list now strictly adheres to American English alphabet constraints, aligning with the app's focus on learning ASL FingerSpelling.

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt [hunk 1]
```diff
@@ -360,20 +364,55 @@ fun ConfirmationDialog(
...
+          Text(
+              text = selectedLanguage,
+              style = MaterialTheme.typography.bodyLarge,
+              color = MaterialTheme.colorScheme.onBackground)
+
+          Icon(
+              imageVector =
+                  if (expanded) Icons.Filled.KeyboardArrowDown
+                  else Icons.AutoMirrored.Filled.KeyboardArrowRight,
+              contentDescription = "Expand language menu",
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:387]: This string can be replaced with a string ressource.

**Review by nada-ezziti — CHANGES_REQUESTED** [review]
This is a solid PR that addresses and fixes several UI issues we've been facing. The changes, such as dynamic header calculation, image border adjustments, and language switch enhancements, significantly improve the user experience and code maintainability. I checked out the branch, and everything is working well as intended, with no noticeable issues.

It would be great to include some screenshots showcasing the updates in your PR description, especially for the Language Switch dropdown and other visual changes, to give a clear idea of the impact of these improvements.

The implementation looks good overall, and I’ll approve the PR once the few suggestions I made are addressed.

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt (hunk 1 above)
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:387]: done in expand_language_menu_text putted in strings.xml

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt [hunk 2]
```diff
@@ -371,3 +371,18 @@ fun QuestsButton(navigationActions: NavigationActions, modifier: Modifier = Modi
...
+ *
+ * @param topBarButtons The list of top bar buttons which are considered as headers.
+ * @param additionalComponents A list of additional composables or header items to count
+ *   dynamically.
+ * @return The total number of headers.
+ */
+fun calculateNumOfHeadersOfTopBarButtons(
+    topBarButtons: List<@Composable () -> Unit>,
+    additionalComponents: List<@Composable () -> Unit>
+): Int {
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:385]: Calculating the number of headers dynamically is definitely a better approach.

**Review by nada-ezziti — APPROVED** [review]
Everything looks good to me, so I approve the PR. Great job!

**Pull request merged**